crates/sui/src/client_commands.rs

```diff
@@ -2641,10 +2641,12 @@ pub(crate) async fn dry_run_or_execute_or_serialize(
         opts.serialize_unsigned_transaction,
         opts.serialize_signed_transaction,
     );
-    assert!(
-        !serialize_unsigned_transaction || !serialize_signed_transaction,
-        "Cannot specify both --serialize-unsigned-transaction and --serialize-signed-transaction"
-    );
+
+    if serialize_signed_transaction && serialize_unsigned_transaction {
+        bail!(
+            "Cannot specify both --serialize-unsigned-transaction and --serialize-signed-transaction."
+        );
+    }
```

Comment:
Note that there is `anyhow::ensure!` that does something similar to assert, but using anyhow:

```suggestion
    ensure!(
        !serialize_unsigned_transaction || !serialize_signed_transaction,
        "Cannot specify both --serialize-unsigned-transaction and --serialize-signed-transaction."
    );
```

Reply:
Ah yes, forgot about ensure!.